features/karaf/openhab-core/src/main/feature/feature.xml

```diff
@@ -155,7 +155,9 @@
 		<feature>openhab-core-auth-jaas</feature>
 		<bundle>mvn:org.openhab.core.bundles/org.openhab.core.io.rest.auth/${project.version}</bundle>
 		<requirement>openhab.tp;filter:="(feature=jose4j)"</requirement>
+		<requirement>openhab.tp;filter:="(feature=swagger-jaxrs)"</requirement>
 		<feature dependency="true">openhab.tp-jose4j</feature>
+		<feature dependency="true">openhab.tp-swagger-jaxrs</feature>
```

Comment:
I'm not too happy with openhab-core-io-rest-auth depending on openhab.tp-swagger-jaxrs after all, if there's a better suggestion (make it optional? bring the ReaderListener, back to org.openhab.core.io.rest.swagger...) I'll gladly take it.

Reply:
Dependency wise I agree it would be better to move `ReaderListener` back to `org.openhab.core.io.rest.swagger`.

Reply:
The intent was to prevent the OAuth2 info injection to occur if auth isn't active i.e. the instance doesn't have the io-rest-auth bundle active. Surely there's a clean way of detecting that differently (and reliably/cleanly) somehow?

Reply:
I'm fine with moving it and always injecting the OAuth2 info for now. We could also create a service that list the available authentication methods so you could query that.